File: payslip-service/src/main/resources/static/js/taxRateService.js
app.service('TaxRateService', function($http) {


 	this.findAll = function() {
 		url = baseurl+"taxRate/findAll";
 		var req = {
 			method : 'GET',
 			url : url,
 			headers : headers
 		}
 		return $http(req);

 	};
 	
 	this.save = function(taxRate) {
 		url = baseurl+"taxRate/save";
 		var req = {
 			method : 'POST',
 			url : url,
 			headers : headers,
 			data : {
 				'taxRate' : taxRate
 			}
 		}
 		return $http(req);

 	};
 	
 	this.delete = function(taxRate) {
 		url = baseurl+"taxRate/delete";
 		var req = {
 			method : 'POST',
 			url : url,
 			headers : headers,
 			data : {
 				'taxRate' : taxRate
 			}
 		}
 		return $http(req);

 	};
 });

File: payslip-service/src/main/java/com/myob/payslip/service/TaxRateService.java
package com.myob.payslip.service;

import java.util.List;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import com.myob.payslip.repository.TaxRate;
import com.myob.payslip.repository.TaxRateRepository;
import com.myob.payslip.util.TaxRateCache;

@Service
public class TaxRateService {
	
	@Autowired
	private TaxRateRepository taxRateRepository;
	
	@Autowired
	private TaxRateCache taxRateCache;
	
	public List<TaxRate> findAll(){
		return taxRateRepository.findAll();
	}
	
	public TaxRate save(TaxRate taxRate){
		taxRate = taxRateRepository.save(taxRate);
		taxRateCache.setTaxRates(taxRateRepository.findAll());
		return taxRate;
	}
	
	public void delete(TaxRate taxRate){
		taxRateRepository.delete(taxRate);
		taxRateCache.setTaxRates(taxRateRepository.findAll());
	}

}
